client/collection_description.go

```diff
@@ -65,6 +65,16 @@ type CollectionDescription struct {
 
 	// Indexes contains the secondary indexes that this Collection has.
 	Indexes []IndexDescription
+
+	// Policy contains the policy information on this collection.
+	//
+	// It is possible for a collection to not have a policy, a collection
+	// without a policy has no access control.
+	//
+	// Note: The policy information must be validated using the acp module
+	// right after parsing is done, to avoid storing an invalid policyID
+	// or policy resource that may not even exist on acp module.
+	Policy immutable.Option[PolicyDescription]
```

Comment:
todo: Please prevent the mutation of this property via `PatchCollection`, as until `https://github.com/sourcenetwork/defradb/issues/2366` is done it could have quite a disruptive effect on an existing collection.

todo: If not done already, please add an error to the P2P config functions so they return a user-friendly error if they try and configure P2P for a collection with a policy.

And integration tests for both tasks linked to #2366.

Reply:
The patch collection point was a good one, since it was merged we have to make sure Policy can't be mutated. Will do.

The rest will also be resolved.

- [ ] prevent patch mutation of Policy

- [ ] error configuring p2p on collection with a policy

- [ ] add tests for the error for prevention of mutation of policy and link with #2366

- [ ] add tests for p2p on collection with a policy and link with #2366

Reply:
Update:

P2P Replicating and Subscribing Disabled in commit: ``` PR(ACP): Disable P2P Replicating & Subscribing ```

Patch Mutation Disabled for Policy in commit: ``` PR(ACP): Disable Mutation Of Policy Through Patch ```